app/models/common/datetime.py

```diff
@@ -31,15 +31,17 @@ class Date(BaseModel):
 
     @validator('day')
     def check_date(cls, v, values):
+        if not 'month' in values:
```

Comment:
I'm not sure if this will work.

You are checking the date against 2016 and 2017 without considering the leap year value at all. Did you test this for leap year versus a normal year?

Reply:
Yes, you can see that she tests for it in the code below this.